src/openrct2/interface/Console.cpp

```diff
@@ -382,16 +400,9 @@ static void console_write_prompt()
 
 void console_write(const utf8 *src)
 {
-    size_t charactersRemainingInBuffer = CONSOLE_BUFFER_SIZE - (_consoleBufferPointer - _consoleBuffer) - 1;
-    size_t charactersToWrite = strlen(src);
-    size_t bufferShift = charactersToWrite - charactersRemainingInBuffer;
-    if (charactersToWrite > charactersRemainingInBuffer) {
-        memmove(_consoleBuffer, _consoleBuffer + bufferShift, CONSOLE_BUFFER_SIZE - bufferShift);
-        _consoleBufferPointer -= bufferShift;
-        charactersRemainingInBuffer = CONSOLE_BUFFER_SIZE - (_consoleBufferPointer - _consoleBuffer) - 1;
-    }
-    safe_strcpy(_consoleBufferPointer, src, charactersRemainingInBuffer);
-    _consoleBufferPointer += charactersToWrite;
+    std::string srcString(src, strlen(src));
+    _consoleVector.insert(_consoleVector.end(), srcString.begin(), srcString.end());
```

Comment:
You could also just use `_consoleVector.push_back()` instead to insert at the end.

Reply:
This line is only adding one string, though, right? The calls to `.begin()` and `.end()` on one `std::string` are a bit misleading in an iterator context, in that respect.

Reply:
`_consoleVector` is a buffer not a list of strings. It needs to copy the chars from the string to the end of the buffer.

Reply:
Yeah, I did try using push_back() but that will only insert the first character of the string. I'll add a comment explaining this though, it isn't obvious.

Reply:
Hmm. Any particular reason why it's not implemented on a line basis? (Or indeed, why we're sticking with it?)

Reply:
Not to my knowledge - I can & will look into that, and removing some of the other C-flavour stuff. For now though this is a mergeable fix, and a step in the right direction at least!